test/export/StructureDefinition.ResourceExporter.test.ts

```diff
@@ -269,12 +269,86 @@ describe('ResourceExporter', () => {
+
+    expect(exported.length).toBe(1);
+    expect(exported[0].elements.filter(r => r.id === 'MyResource.extension')).toHaveLength(1);
+
+    expect(exported[0].name).toBe('MyResource');
+    expect(exported[0].id).toBe('MyResource');
+    expect(exported[0].type).toBe('MyResource');
+    expect(exported[0].baseDefinition).toBe(
+      'http://hl7.org/fhir/StructureDefinition/DomainResource'
+    );
+    expect(exported[0].elements).toHaveLength(9);
+
+    expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+    expect(loggerSpy.getLastMessage('error')).toMatch(
+      `Cannot define element ${addElementRule.path} on ${resource.name} because it has already been defined`
+    );
+  });
+
+  it('should log an error when two rules add a new element with the same path', () => {
+    const resource = new Resource('MyResource');
+    const addElementRule = new AddElementRule('testElement');
+    addElementRule.min = 0;
+    addElementRule.max = '1';
+    const addElementRule2 = new AddElementRule('testElement');
+    addElementRule2.min = 0;
+    addElementRule2.max = '1';
+    resource.rules.push(addElementRule);
+    resource.rules.push(addElementRule2);
+    doc.resources.set(resource.name, resource);
+    const exported = exporter.export().resources;
+
+    expect(exported.length).toBe(1);
+    expect(exported[0].elements.filter(r => r.id === 'MyResource.testElement')).toHaveLength(1);
+
+    expect(exported[0].name).toBe('MyResource');
+    expect(exported[0].id).toBe('MyResource');
+    expect(exported[0].type).toBe('MyResource');
+    expect(exported[0].baseDefinition).toBe(
+      'http://hl7.org/fhir/StructureDefinition/DomainResource'
+    );
+    expect(exported[0].elements).toHaveLength(10);
+
+    expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+    expect(loggerSpy.getLastMessage('error')).toMatch(
+      `Cannot define element ${addElementRule.path} on ${resource.name} because it has already been defined`
+    );
+  });
+
+  it('should log an error when a rule with the same path is added by directly calling newElement', () => {
+    doc = new FSHDocument('fileName');
+    const input = new FSHTank([doc], minimalConfig);
+    pkg = new Package(input.config);
+    const fisher = new TestFisher(input, defs, pkg);
+    const alternateIdentification: StructureDefinition =
+      fisher.fishForStructureDefinition('AlternateIdentification');
+
+    try {
+      const addElementRule = new AddElementRule('elem1');
+      addElementRule.min = 0;
+      addElementRule.max = '1';
+      const addElementRule2 = new AddElementRule('elem1');
+      addElementRule2.min = 0;
+      addElementRule2.max = '1';
+      alternateIdentification.newElement(addElementRule.path);
+      alternateIdentification.newElement(addElementRule2.path);
+    } catch (e) {
+      expect(e.message).toMatch(
+        'Cannot define element elem1 on AlternateIdentification because it has already been defined'
+      );
+    }
```

Comment:
I think there is a little cleaner of a way to test that the error is thrown. This is following another pattern we have used in other tests, so it might make sense to use this approach for consistence.

Instead of wrapping everything in a try/catch statement, you can wrap the second `newElement` in a function and use a `expect(...).toThrow(...)` instead of the try/catch. So in this case, you can do this:

```
expect(() => {
  alternateIdentification.newElement(addElementRule2.path);
}).toThrow(
  'Cannot define element elem1 on AlternateIdentification because it has already been defined'
);
```

Once you do that, you don't need the try/catch block anymore.

There are various examples of this in the code, but you can see one of them [here](https://github.com/FHIR/sushi/blob/master/test/fhirtypes/ElementDefinition.assignBoolean.test.ts#L59-L77).

Reply:
Done, thanks!